### ISSUE TITLE
homebrew_tap untap action does not work on Apple Silicon. 

### ISSUE THREAD

**Issue by weswhet** [chef/chef#13721]
## Description

When using the homebrew_tap resource on Apple Silicon chef will not remove the requested tap. This is because it is looking for a [specific path in homebrew](https://github.com/chef/chef/blob/main/lib/chef/resource/homebrew_tap.rb#L79) to see if the tap is installed, since the tap doesn't exist chef thinks it's already been removed. Unfortunately this path is not correct on Apple Silicon. brew was moved to `/opt/homebrew`. Ideally the code should check `brew tap` for a list of all homebrew taps. This would allow chef to manage homebrew regardless of device architecture and install location.
## Platform Version

macOS 13.3.1 Apple Silicon

## Replication Case

`brew tap hashicorp/tap`

```
homebrew_tap 'hashicorp/tap' do
  homebrew_path      "/opt/homebrew/bin/brew"
  action             :untap
end
```

**Comment by jaymzh** [comment]
There's a PR for this already #13669

**Comment by tpowell-progress** [comment]
Fixed completed with 18.3.0 (which had a follow-up to #13669)
